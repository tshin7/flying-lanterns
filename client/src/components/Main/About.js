import React, { Component } from 'react';

export default class About extends Component {
  render() {
    const {mainStyle} = this.props;
    return (
      <div style={mainStyle}>
        Flying Lanterns provides costumes and accessories for cosplaying.
        At the moment. everything on this site is created by Misha.
      </div>
    );
  }
}
