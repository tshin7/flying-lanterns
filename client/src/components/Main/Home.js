import React, { Component } from 'react';
import { Main } from './Main.style.js';

export default class Home extends Component {
  render() {
    return (
      <Main>
        Home
      </Main>
    );
  }
}
