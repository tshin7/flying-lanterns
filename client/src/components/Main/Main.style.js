import styled from 'styled-components';

export const Main = styled.div`
  marginTop: 4em;
  textAlign: center;
  marginBottom: 50em;
`;
