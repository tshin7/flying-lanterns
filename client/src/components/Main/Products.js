import React, { Component } from 'react';

export default class Products extends Component {
  render() {
    const {mainStyle} = this.props;
    return (
      <div style={mainStyle}>
        Products
      </div>
    );
  }
}
