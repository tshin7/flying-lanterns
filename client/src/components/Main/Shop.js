import React, { Component } from 'react';
import { MainDiv } from './Main.style.js';

export default class Products extends Component {
  render() {
    return (
      <MainDiv>
        Products
      </MainDiv>
    );
  }
}
