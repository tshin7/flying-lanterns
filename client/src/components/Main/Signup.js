import React, { Component } from 'react';

export default class Signup extends Component {
  render() {
    const {mainStyle} = this.props;
    return (
      <div style={mainStyle}>
        Signup
      </div>
    );
  }
}
