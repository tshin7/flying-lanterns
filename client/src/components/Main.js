import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from './Main/Home';
import Shop from './Main/Shop';
import About from './Main/About';
import Login from './Main/Login';
import Signup from './Main/Signup';

const Main = () => (
  <main>
    <Switch>
      <Route
        exact
        path="/"
        render={props => <Home {...props} />}
      />
      <Route
        exact
        path="/shop"
        render={props => <Shop {...props} />}
      />
      <Route
        exact
        path="/about"
        render={props => <About {...props} />}
      />
      <Route
        exact
        path="/login"
        render={props => <Login {...props} />}
      />
      <Route
        exact
        path="/signup"
        render={props => <Signup {...props} />}
      />
    </Switch>
  </main>
);

export default Main;
