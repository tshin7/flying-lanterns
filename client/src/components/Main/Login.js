import React from 'react';
import {
  Button, Form, Grid, Header, Message, Segment,
} from 'semantic-ui-react';
import { MainDiv } from './Main.style';

const Login = () => (
  <MainDiv>
    <Grid textAlign="center" style={{ height: '100%' }} verticalAlign="middle">
      <Grid.Column style={{ maxWidth: 450 }}>
        <Header as="h2" color="teal" textAlign="center">
          Login to your account
        </Header>
        <Form size="large">
          <Segment stacked>
            <Form.Input fluid icon="user" iconPosition="left" placeholder="E-mail address" />
            <Form.Input
              fluid
              icon="lock"
              iconPosition="left"
              placeholder="Password"
              type="password"
            />
            <Button color="teal" fluid size="large">
              Login
            </Button>
          </Segment>
        </Form>
        <Message>
          New to us?
          {' '}
          <a href="/signup">Sign Up</a>
        </Message>
      </Grid.Column>
    </Grid>
  </MainDiv>
);

export default Login;
