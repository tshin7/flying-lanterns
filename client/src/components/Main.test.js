import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

describe('Main', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<Main />);

    expect(component).toMatchSnapshot();
  });

});
