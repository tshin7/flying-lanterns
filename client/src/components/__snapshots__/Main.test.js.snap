// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly with no props 1`] = `
<main>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/shop"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/about"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/signup"
      render={[Function]}
    />
  </Switch>
</main>
`;
